txpool/txpool_test.go

```diff
@@ -3709,14 +3711,55 @@ func TestBatchTx_SingleAccount(t *testing.T) {
 	// mutex for txHashMap
 	mux := &sync.RWMutex{}
 	counter := uint64(0)
-
+	enqueuedCount := 0
+	promotedCount := 0
+	ev := (*proto.TxPoolEvent)(nil)
 	wg := sync.WaitGroup{}
-	wg.Add(int(defaultMaxAccountEnqueued))
+	wg.Add(1)
+
+	// wait for all the submitted transactions to be promoted
+	go func() {
+		defer wg.Done()
+
+		for {
+			select {
+			case ev = <-subscription.subscriptionChannel:
+			case <-time.After(time.Second * 6):
+				t.Fatalf("timeout. processed: %d/%d and %d/%d. Added: %d",
+					enqueuedCount, defaultMaxAccountEnqueued, promotedCount, defaultMaxAccountEnqueued,
+					atomic.LoadUint64(&counter))
+			}
+
+			// check if valid transaction hash
+			mux.Lock()
+			_, hashExists := txHashMap[types.StringToHash(ev.TxHash)]
+			mux.Unlock()
```

Comment:
This thread is only one reader, so this will not have any effect in performances. However it is more appropriate to do it this way so will add this suggestion.